### PR TITLE
[Flax MLM] Refactor run mlm with optax

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/flax/language-modeling/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+### Train tokenizer
+
+In a first step, we train a tokenizer to efficiently process the text input for the model. Similar to how it is shown in [How to train a new language model from scratch using Transformers and Tokenizers](https://huggingface.co/blog/how-to-train), we use a **`ByteLevelBPETokenizer`**.
+The tokenizer is trained on the complete Norwegian dataset of OSCAR
+and consequently saved in `${MODEL_DIR}`
+
+```python
+from datasets import load_dataset
+from tokenizers import trainers, Tokenizer, normalizers, ByteLevelBPETokenizer
+from transformes import RobertaFastTokenizer
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:57]:
```suggestion
```

File: examples/flax/language-modeling/README.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+In the following, we demonstrate how to train a bi-directional transformer model 
+using masked language modeling objective as introduced in [BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding](https://arxiv.org/abs/1810.04805).
+More specifically, we demonstrate how JAX/Flax can be leveraged 
+to pre-train [**`roberta-base`**](https://huggingface.co/roberta-base)
+in Norwegian on a single TPUv3-8 pod.
+
+The example script uses the 🤗 Datasets library. You can easily customize them to your needs if you need extra processing on your datasets.
+
+Let's start by creating a folder to save the trained model.
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:41]:
```suggestion
Let's start by creating a folder to save the trained model and link to the `run_mlm_flax.py` script.
```

File: examples/flax/language-modeling/README.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+```bash
+export MODEL_DIR="./norwegian-roberta-base"
+mkdir -p ${MODEL_DIR}
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:45]:
```suggestion
mkdir -p ${MODEL_DIR}
ln -s ~/transformers/examples/flax/language-modeling/run_mlm_flax.py run_mlm_flax.py
```

File: examples/flax/language-modeling/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        --weight_decay="0.01" \
+        --per_device_train_batch_size="128" \
+        --per_device_eval_batch_size="128" \
+        --learning_rate="3e-4" \
+        --warmup_steps="1000" \
+        --overwrite_output_dir \
+        --pad_to_max_length \
+        --num_train_epochs="18" \
+        --adam_beta1="0.9" \
+        --adam_beta2="0.98" \
```
Comment (patrickvonplaten) [inline examples/flax/language-modeling/README.md:124]:
```suggestion
        --adam_beta2="0.98"
```
